### DISCUSSION TITLE
How to build  retracting side bar / drawer (eg. mini-mode)

### DISCUSSION THREAD

**Discussion by AdamC03** [zauberzeug/nicegui#610] [Q&A] [answered]
It may already be in nice gui but I couldn't find it in the documentation. I know I cam add custom html/css however I don't know how to even begin implementing something like this

https://i.imgur.com/fZFVsQb.png --> https://i.imgur.com/Xk8RegS.png

**Comment by rodja** [comment, marked as answer]
I think it's the [QDrawer mini mode](https://quasar.dev/layout/drawer) you are looking for:

```py
def toggle():
    if 'mini' in drawer._props:
        drawer.props(remove='mini')
    else:
        drawer.props('mini')


with ui.left_drawer(top_corner=True, bottom_corner=True).props('mini').classes('bg-primary p-0') as drawer:
    ui.button(on_click=toggle).props('icon=menu flat color=white').classes('pt-4')
```

**Reply by AdamC03** [reply]
Ill try this out and ill lyk of the results, thanks for the continued support <3

**Reply by AdamC03** [reply]
That solution works great but I noticed a small issue with the window sizes, it seems to have a minimum window size to work, anything below that it seems to bug out and become un-usable. I made a little gif displaying this behavior: https://imgur.com/a/aUUKZCJ
Exact code:
```py
from nicegui import ui
def toggle():
    if 'mini' in drawer._props:
        drawer.props(remove='mini')
    else:
        drawer.props('mini')

with ui.left_drawer(top_corner=True, bottom_corner=True).props('mini').classes('bg-primary p-0') as drawer:
    ui.button(on_click=toggle).props('icon=menu flat color=white').classes('pt-4')
ui.run(dark=True)
```

**Reply by falkoschindler** [reply]
This is intended behavior. Depending on the `breakpoint` the layout automatically switches from `desktop` to `mobile`. But you can enforce keeping the drawer visible using `behavior=desktop`:
```py
with ui.left_drawer(...).props('mini behavior=desktop') ...
```
See [Quasar's Drawer API](https://quasar.dev/layout/drawer#qdrawer-api) for more information.

**Reply by AdamC03** [reply]
Thank you very much, you guys have been absolute madlads
